bodo/io/iceberg/catalog/snowflake.py

```diff
@@ -320,3 +364,25 @@ def list_views(self, namespace: str | Identifier) -> list[Identifier]:
 
     def drop_view(self, identifier: str | Identifier) -> None:
         raise NotImplementedError
+
+    def prefetch_metadata_paths(self, table_ids: pt.Sequence[str | Identifier]) -> None:
+        """
+        For a given list of tables (by their identifiers), start the initial
+        fetch for the head metadata file for each table by submitting an async
+        SQL query through the Snowflake connector.
+
+        During load_table, if the catalog sees an open metadata file request,
+        it will block for the async query to complete and use the result.
+        """
+
+        for table_id in table_ids:
+            sf_identifier = SnowflakeCatalog._SnowflakeIdentifier.table_from_id(
+                table_id
+            )
+
+            with self.connection.cursor(DictCursor) as cursor:
+                cursor.execute_async(METADATA_QUERY, (sf_identifier.table_name,))
+                query_id = cursor.sfqid
+                assert query_id is not None
+
+            self.open_metadata_requests[sf_identifier.table_name] = query_id
```

Comment:
Do we need the whole table id to avoid name conflicts in the dictionary?

Reply:
`sf_identifier.table_name` is actually the whole table_id unescaped. Was part of the original code, dont love the naming. Something we should change if/when we open up a PR on PyIcebergs side